piezo_web_app/PiezoWebApp/src/services/spark_job/spark_job_service.py

```diff
@@ -181,6 +182,37 @@ def submit_job(self, body):
                 'message': message
             }
 
+    def tidy_jobs(self):
+        api_response = self.get_jobs(label=None)
+        if api_response['status'] != StatusCodes.Okay.value:
+            return api_response
+        dict_of_jobs = api_response['spark_jobs']
+        jobs_untouched = {}
+        jobs_tidied = {}
+        jobs_failed_to_process = {}
+
+        for job in dict_of_jobs:
+            status = dict_of_jobs[job]
+            if status in ["COMPLETED", "FAILED"]:
+                write_logs_response = self.write_logs_to_file(job)
+                if write_logs_response['status'] == StatusCodes.Okay.value:
+                    delete_response = self.delete_job(job)
+                    if delete_response['status'] == StatusCodes.Okay.value:
+                        jobs_tidied[job] = status
+                    else:
+                        jobs_failed_to_process[job] = delete_response['message']
+                else:
+                    jobs_failed_to_process[job] = write_logs_response['message']
+
+            else:
+                self._logger.debug(f'Not processing job "{job}", current status is "{status}"')
+                jobs_untouched[job] = status
+        return {'status': StatusCodes.Okay.value,
```

Comment:
Should the status always be okay? E.g. if all jobs fail to process?

Reply:
Agreed this case probably shouldn't be 200. Should I change it if any fail, maybe to a 412 Precondition Failed

Reply:
Some but not all of the tasks failing seems a lot less likely than all of them failing (if there's an underlying problem then it should hit all of them). In the event that only some fail then they'll show up in the `jobs_failed_to_process` part of the response.

Reply:
After discussion we have left the status code alone due to complications with potentially different status code being returned. Instead we have extended the output of failed jobs to include it's error message and status code of the failure